### PR TITLE
Add version info to manager and userAgent in Tilt

### LINKED ISSUE #1949: user agent CAPZ version is missing
/kind bug

[Before submitting an issue, have you checked the [Troubleshooting Guide](https://capz.sigs.k8s.io/topics/troubleshooting.html)?]

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]

Using v1.1.0 tag, seeing user agent as `Go/go1.16.12 (amd64-linux) go-autorest/v14.2.1 Azure-SDK-For-Go/v58.1.0 network/2021-02-01 cluster-api-provider-azure/`

Screen Shot 2022-01-05 at 4 49 37 PM

**What did you expect to happen:**

The user agent should have the Git Version of capz, ie. `cluster-api-provider-azure/v1.1.0`.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- cluster-api-provider-azure version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by Jont828**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**: The version info added in #639 is not passed in when CAPZ is started using Tilt.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1949 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Add version info to Tilt
```

### CODE REVIEW

**Comment by mboersma** [comment]
I also noticed this bug [back in July](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1498#pullrequestreview-701338920). Thanks for tracking it down!

**Review by CecileRobertMichon — COMMENTED** [review]
lgtm

As discussed with @Jont828 offline, he's going to look into why the same issue is happening with the published manager image built with `make docker-build` and try to fix both in this PR.

**Comment by CecileRobertMichon** [comment]
/lgtm

@Jont828 I'll let you confirm the output in the e2e test is what you want before we merge this

**Comment by Jont828** [comment]
Sounds good, I think I might need to rerun the tests b/c of flakes

File: Makefile [hunk 1]
```diff
@@ -224,6 +224,18 @@ endif
 $(KUBE_APISERVER) $(ETCD): ## install test asset kubectl, kube-apiserver, etcd
 	source ./scripts/fetch_ext_bins.sh && fetch_tools
 
+.PHONY: env-info # Temporary target to get additional logs in prow tests
+env-info:
+	@echo "Working dir is $(shell pwd)"
+	@echo "Working dir contents:"
+	ls -la
+	@echo "Root dir contents:"
+	ls -la /
```
Comment (CecileRobertMichon) [inline Makefile:233]: /hold

as discussed offline, please add an `|| echo ""` to avoid failing the job if we are unable to list root

**Comment by Jont828** [comment]
@CecileRobertMichon I fixed the changes and it looks like we were able to get the outputs in the log. Does this look good to merge?

**Comment by Jont828** [comment]
@CecileRobertMichon I think these changes should do it!

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm

**Pull request merged**